platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/PkiTools.java

```diff
@@ -243,13 +268,12 @@ public static X500Name makeDistinguishedName(String commonName) {
     public static X509Certificate pemToCertificate(String certString) {
         CertificateFactory cf = new CertificateFactory();
         ByteArrayInputStream in = new ByteArrayInputStream(PkiTools.pemToDer(certString));
-        X509Certificate cert = null;
+        X509Certificate cert;
```

Comment:
I think I see what's going on. It looks like the compiler only initializes fields on the object. Local vars are not. I had a peek at the Java Language Spec and found:

_For every access of a local variable or blank final field x, x must be definitely assigned before the access, or a compile-time error occurs._

I guess `cert` is _definitely assigned_ because if the try-with-resources fails, the catch/throw interrupts the control flow and the value of `cert` is never used.